### PR TITLE
Dcp 646: Return null when submission is empty

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#646
ebi-ait/dcp-ingest-central#600

### CODE REVIEW

**Review by jacobwindsor — APPROVED** [review]

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
 import org.springframework.retry.support.RetryTemplate;
 import org.springframework.stereotype.Service;
 
+import javax.swing.text.html.Option;
```
Comment (amnonkhen) [inline src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java:34]: Is this a correct import?

File: src/main/java/org/humancellatlas/ingest/submission/web/SubmissionController.java [hunk 2]
```diff
@@ -365,7 +365,7 @@
 
     @GetMapping("/submissionEnvelopes/{id}" + Links.SUBMISSION_CONTENT_LAST_UPDATED_URL)
     ResponseEntity<?> getContentLastUpdated(@PathVariable("id") SubmissionEnvelope submissionEnvelope) {
-        Instant lastUpdateDate = submissionEnvelopeService.getSubmissionContentLastUpdated(submissionEnvelope);
-        return ResponseEntity.ok(lastUpdateDate.toString());
+        Optional<Instant> lastUpdateDate = submissionEnvelopeService.getSubmissionContentLastUpdated(submissionEnvelope);
+        return ResponseEntity.ok(lastUpdateDate.isPresent() ? lastUpdateDate.get().toString() : null);
```
Comment (amnonkhen) [inline src/main/java/org/humancellatlas/ingest/submission/web/SubmissionController.java:369]: Have you considered
```suggestion
        return ResponseEntity.ok(lastUpdateDate.map(Instant::toString).orElse(null));
```

Comment (amnonkhen) [inline src/main/java/org/humancellatlas/ingest/submission/web/SubmissionController.java:369]: What does the null case mean here?
Would it be correct to have an `ok(null)` return value?

File: src/test/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeServiceTest.java [hunk 3]
```diff
@@ -356,10 +356,35 @@ public void testContentLastUpdated() {
         when(file.getUpdateDate()).thenReturn(now);
 
         // when
-        Instant lastUpdateDate = service.getSubmissionContentLastUpdated(submission);
+        Optional<Instant> lastUpdateDate = service.getSubmissionContentLastUpdated(submission);
 
         // then
-        assertThat(lastUpdateDate).isEqualTo(now.toString());
+        assertThat(lastUpdateDate.isPresent()).isTrue();
+        assertThat(lastUpdateDate.get().toString()).isEqualTo(now.toString());
```
Comment (amnonkhen) [inline src/test/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeServiceTest.java:363]: AssertJ handles Optionals elegantly.
See https://www.baeldung.com/assertJ-java-8-features
```suggestion
        assertThat(lastUpdateDate)
            .isPresent()
            .hasValue(now.toString());
```

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**